sql/sql_plugin.cc

```diff
@@ -472,7 +492,10 @@ static st_plugin_dl *plugin_dl_add(const LEX_STRING *dl, int report)
                                system_charset_info, 1) ||
       plugin_dir_len + dl->length + 1 >= FN_REFLEN)
   {
-    report_error(report, ER_UDF_NO_PATHS);
+    *err= ER_UDF_NO_PATHS;
```

Comment:
Good point. The reason is initially I wanted to add the ability to lookup plugin until it can be loaded, so there could be several plugin libs with load error and this error should be handled by the up-level function. But now the lookup stops on first plugin file load error, so there is no need to pass error string to caller.